WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderSiteHeaderView.java

```diff
@@ -237,18 +193,18 @@ private void toggleFollowStatus(final View followButton) {
         }
 
         final boolean isAskingToFollow;
-        if (mFeedId != 0) {
+        if (mIsFeed) {
             isAskingToFollow = !ReaderBlogTable.isFollowedFeed(mFeedId);
         } else {
             isAskingToFollow = !ReaderBlogTable.isFollowedBlog(mBlogId);
         }
 
         if (mFollowListener != null) {
-            if (isAskingToFollow) {
+            if (isAskingToFollow && !mIsFeed) {
```

Comment:
Could you please elaborate on this condition - it seems a bit suspicious -> the user clicks on `follow` and we perform `unfollow` action when it's a feed?

Reply:
Ash, I see the confusion. The listener handlers don't do any server logic, the followed one will dispatches a newFetchSubscriptionsAction and show a snackbar (which we don't want to show in the case of feed). The isFollowing just dispatches a newFetchSubscriptionsAction. I adjusted so it goes down the same path, but will not show a snackbar if blog is not > 0. Easier to understand. Adjusted in e59a23b